corehq/apps/toggle_ui/views.py

```diff
@@ -94,10 +111,14 @@ def post(self, request, *args, **kwargs):
             item_list = json.loads(item_list)
             item_list = [u.strip() for u in item_list if u and u.strip()]
 
-        affected_users = set(toggle.enabled_users) | set(item_list)
+        old_entries = set(toggle.enabled_users)
+        new_entries = set(item_list)
         toggle.enabled_users = item_list
         toggle.save()
-        for item in affected_users:
+
+        changed_entries = old_entries ^ new_entries  # ^ is XOR in python
+        self.call_save_fn(changed_entries, new_entries)
+        for item in changed_entries:
             clear_toggle_cache(toggle.slug, item)
```

Comment:
Note that this is a change in the cache invalidation logic too - previously, `clear_toggle_cache` was called on the all the old and new entries.  This changes it to clear the cache only for those entries whose toggle status has changed (they're either in the old set or the new one, but not both).